pkcs11/_pkcs11.pyx

```diff
@@ -377,33 +382,73 @@ cdef class MechanismWithParam:
         PyMem_Free(self.param)
 
 
-class Slot(types.Slot):
+cdef class Slot(HasFuncList, types.Slot):
     """Extend Slot with implementation."""
 
+    cdef readonly CK_SLOT_ID slot_id
+    """Slot identifier (opaque)."""
+    cdef readonly str slot_description
+    """Slot name (:class:`str`)."""
+    cdef readonly str manufacturer_id
+    """Slot/device manufacturer's name (:class:`str`)."""
+    cdef readonly tuple cryptoki_version
+    cdef CK_FLAGS slot_flags
+    cdef CK_VERSION hw_version
+    cdef CK_VERSION fw_version
+
+    @staticmethod
+    cdef Slot make(CK_FUNCTION_LIST *funclist, CK_SLOT_ID slot_id, CK_SLOT_INFO info, tuple cryptoki_version):
```

Comment:
Using Cython extension types (`cdef` classes) pretty much forces moving a bunch of state-touching logic from `types.py` to the Cython layer. It's not too bad, though. API-wise, there's no real breakage for a user that doesn't access any of the Cython internals directly.